pkg/k8shandler/cluster.go

```diff
@@ -47,183 +46,150 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
-					// perform a full cluster update
-					if err := elasticsearchRequest.PerformFullClusterUpdate(scheduledNodes); err != nil {
-						logrus.Warnf("when trying to perform full cluster update: %s", err.Error())
-					}
+		// if it is < what we expect (6.0) then do full cluster update:
+		if comparison > 0 {
+			// perform a full cluster update
+			if err := er.PerformFullClusterUpdate(scheduledNodes); err != nil {
+				logrus.Warnf("when trying to perform full cluster update: %s", err.Error())
+				return er.UpdateClusterStatus()
+			}
 
-				} else {
+		} else {
 
-					if err := elasticsearchRequest.PerformRollingUpdate(scheduledNodes); err != nil {
-						logrus.Warnf("when trying to perform rolling update: %s", err.Error())
-					}
-				}
+			if err := er.PerformRollingUpdate(scheduledNodes); err != nil {
+				logrus.Warnf("when trying to perform rolling update: %s", err.Error())
+				return er.UpdateClusterStatus()
 			}
+		}
 
-		} else {
-			// FIXME: either add logic that will do just restarts of nodes or remove this code
-			// Check if we have any nodes scheduled for a restart
-			scheduledRestartNodes := getScheduledRedeployOnlyNodes(elasticsearchRequest.cluster)
-			if len(scheduledRestartNodes) > 0 {
-				// get all nodes that need only a restart
-
-				if err := elasticsearchRequest.PerformRollingRestart(scheduledRestartNodes); err != nil {
-					logrus.Warnf("when trying to perform rolling restart: %v", err)
-				}
-			} else {
+		er.UpdateClusterStatus()
+	}
 
-				// We have no updates or restarts scheduled
-				// create any nodes we are missing and perform any required operations to ensure state
-				for _, node := range nodes[nodeMapKey(elasticsearchRequest.cluster.Name, elasticsearchRequest.cluster.Namespace)] {
-					clusterStatus := elasticsearchRequest.cluster.Status.DeepCopy()
-					_, nodeStatus := getNodeStatus(node.name(), clusterStatus)
-
-					// Verify that we didn't scale up too many masters
-					if err := elasticsearchRequest.isValidConf(); err != nil {
-						// if wrongConfig=true then we've already print out error message
-						// don't flood the stderr of the operator with the same message
-						if wrongConfig {
-							return nil
-						}
-						wrongConfig = true
-						return err
-					}
+	// Check if we have any nodes scheduled for a restart
+	scheduledRestartNodes := er.getScheduledRedeployOnlyNodes()
+	if len(scheduledRestartNodes) > 0 {
+		// get all nodes that need only a restart
 
-					if err := node.create(); err != nil {
-						return err
-					}
+		if err := er.PerformRollingRestart(scheduledRestartNodes); err != nil {
+			logrus.Warnf("when trying to perform rolling restart: %v", err)
+			return er.UpdateClusterStatus()
+		}
 
-					addNodeState(node, nodeStatus)
-					if err := elasticsearchRequest.setNodeStatus(node, nodeStatus, clusterStatus); err != nil {
-						logrus.Warnf("unable to set status for node %q: %s", node.name(), err.Error())
-					}
+		er.UpdateClusterStatus()
```

Comment:
No return here? Or is this the same thing to the `setNodeStatus` + `Warnf` on the [left L156](https://github.com/openshift/elasticsearch-operator/pull/424/files#diff-b0f11234653f638f62ef2fc4f695a69dL156)?

Reply:
no return because we were able to complete the restarts -- we fall through to the next list of possible actions for the operator. the status update is our CR status is more current